### ISSUE TITLE
Backslash escaped IRI local names are not parsed

### ISSUE THREAD

**Issue by wouterbeek** [drobilla/serd#13]
I'm posting here an issue related to an issue opened by @wonkydonky in the hdt-cpp project.

IIUC it is allowed to use backslash escaping in IRI local names, which would allow dots to appear as the first and/or last character of a local name (something that is not allowed without backslash escaping).

A simple example to illustrate this:

```
$ serdi -s "prefix : <x:x> :\. : : ."
error: (string):1:20: bad subject
```

**Comment by drobilla** [comment]
Sigh, yes, this is allowed in RDF 1.1 Turtle, for a weird specific set of characters that has nothing to do with Turtle and is only there because the SPARQL grammar is horribly designed and not extensible at all.  Bonus points for grafting this on at the last minute and not actually updating the test suite to cover any of it.  All of that and we still can't use slashes in local names.  Concatenating strings should _really_ not be this complicated!

Complaining aside, I will try to write a test suite and implement this mess when I get some time.

**Comment by drobilla** [comment]
This is specifically to do with the special (nightmare to parse) case of a trailing dot in a local name.

**Comment by drobilla** [comment]
Fixed in f124a07
